From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Juan Sanchez Fuentes <juanesf91@gmail.com>
Date: Thu, 14 May 2026 21:10:49 -0400
Subject: Update WiFi node patch on Radxa Cubie A5E

Adapted from Armbian build@0648ff3c4125d673c18b5f032dc7c28545c542b5,
patch/kernel/archive/sunxi-6.18/patches.backports/22-Enable-wifi-on-Radxa-Cubie-A5E.patch.
Keep the kernel and firmware descriptions paired: Linux normally uses
the firmware DT to preserve SID-derived Ethernet MAC addresses.

Signed-off-by: Juan Sanchez Fuentes <juanesf91@gmail.com>
---
 dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts | 34 ++++++++++
 1 file changed, 34 insertions(+)

diff --git a/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts b/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts
index 111111111111..222222222222 100644
--- a/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts
+++ b/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts
@@ -47,6 +47,21 @@ iio-hwmon {
 			      <&axp717_adc 4>; /* pmic_temp */
 	};
 
+	reg_3v3_wifi: 3v3-wifi {
+		compatible = "regulator-fixed";
+		regulator-name = "3v3-wifi";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&r_pio 0 7 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	wifi_pwrseq: wifi-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		reset-gpios = <&r_pio 1 1 GPIO_ACTIVE_LOW>;
+		post-power-on-delay-ms = <200>;
+	};
+
 	reg_vcc5v: vcc5v {
 		/* board wide 5V supply from the USB-C connector */
 		compatible = "regulator-fixed";
@@ -129,6 +144,25 @@ &mmc0 {
 	status = "okay";
 };
 
+&mmc1 {
+	bus-width = <4>;
+	max-frequency = <40000000>;
+	mmc-pwrseq = <&wifi_pwrseq>;
+	non-removable;
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc1_pins>;
+	vmmc-supply = <&reg_3v3_wifi>;
+	vqmmc-supply = <&reg_bldo1>;
+	status = "okay";
+
+	wifi@1 {
+		reg = <1>;
+		interrupt-parent = <&r_pio>;
+		interrupts = <1 0 IRQ_TYPE_LEVEL_LOW>;
+		interrupt-names = "host-wake";
+	};
+};
+
 &ohci0 {
 	status = "okay";
 };
-- 
Armbian

